opennms-webapp-rest/src/test/java/org/opennms/web/rest/v2/MonitoringLocationRestServiceIT.java

```diff
@@ -164,4 +172,68 @@ public void testEventOnCreationAndDeletion() throws Exception {
         // delete the one with polling packages
         sendRequest(DELETE, "/monitoringLocations/location2", 204);
     }
+
+    @Test
+    @Transactional
+    public void testLocationLimits() throws Exception {
+        // There is one default location
+        for (int i = 0; i < LOCATION_COUNT - 1; i++) {
+            Location loc = new Location();
+            loc.locationName = "Location-" + String.format("%05d", i);
+            loc.monitoringArea = "LocationArea-" + String.format("%05d", i);
+
+            // post it to web service
+            String location = "<location location-name=\"" + loc.locationName + "\" monitoring-area=\"" + loc.monitoringArea + "\"/>";
+            sendPost("/monitoringLocations", location, 201, null);
+        }
+
+        // Map contains (limit, expected count)
+        Map<Integer,Integer> limits = new HashMap<Integer,Integer>(8);
+        limits.put(null,DEFAULT_LIMIT); // null limit, should return default
+        limits.put(DEFAULT_LIMIT - 1,DEFAULT_LIMIT - 1); // limit less than default limit
+        limits.put(DEFAULT_LIMIT,DEFAULT_LIMIT); // limit equals default
+        limits.put(DEFAULT_LIMIT + 1,DEFAULT_LIMIT + 1); // limit greater than default
+        limits.put(LOCATION_COUNT + 1,LOCATION_COUNT); // limit greater than count
+        limits.put(0,LOCATION_COUNT); // unlimited, should return all
```

Comment:
IMO this should be flatten to a single call per test. Something like
```
assertThat(getJsonObject(MAPPER, "/monitoringLocations", Map.of("limit", 0), OnmsMonitoringLocationDefinitionList.class).count, is(15));
```

Reply:
Ok, I will do that